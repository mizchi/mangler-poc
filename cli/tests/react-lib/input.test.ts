import path from "node:path";
import fs from "node:fs";
import { bundle, execBuild } from "../_testUtils";
import { beforeAll, expect, test } from "vitest";

const cwd = path.dirname(new URL(import.meta.url).pathname);

let analyzed: any;
beforeAll(async () => {
  await execBuild(cwd, { builtins: ["es", "react"] });
  analyzed = JSON.parse(fs.readFileSync(path.join(cwd, "analyzed.json"), "utf-8"));
});

test("compile with analyze", async () => {
  const fileName = path.join(cwd, "input.tsx");
  const out = await bundle(
    fileName,
    {
      mangle: {
        properties: {
          builtins: true,
          regex: /.*/,
          reserved: analyzed.reserved,
        },
      },
    },
    ["react", "react/jsx-runtime", "react/jsx-dev-runtime"],
  );
  expect(out).includes("display");
  expect(out).includes("justifyContent");
  expect(out).includes("alignItems");
  expect(out).includes("children");
  expect(out).not.includes("internal");
  expect(out).toMatchSnapshot();
});
