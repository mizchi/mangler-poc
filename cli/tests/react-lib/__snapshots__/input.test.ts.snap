// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile with analyze 1`] = `
"import{jsxs as n,jsx as e}from\\"react/jsx-runtime\\";import{useRef as r}from\\"react\\";function t(t){const c=r(null);return n(\\"div\\",{ref:c,style:{display:\\"flex\\",justifyContent:\\"center\\",alignItems:\\"center\\"},children:[e(\\"h1\\",{children:t.title}),e(i,{n:42})]})}function i(n){return e(\\"span\\",{children:n.n})}export{t as MyApp};
"
`;
